src/components/shared/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+
+import NumberInputContainer from './NumberInput.container';
+import Fab from 'components/shared/Fab';
+
+interface Props {
+  value: number;
+  max: number;
+  min: number;
+  onChange: (value: number) => void;
+  step: number;
+}
+
+interface State {
+  value: number;
+}
+
+class NumberInput extends React.Component<Props, State> {
+  static defaultProps: Props = {
+    value: 0,
+    max: Number.MAX_SAFE_INTEGER,
+    min: Number.MIN_SAFE_INTEGER,
```

Comment:
should the default min be `0` instead?

Reply:
ya `0` makes sense for practical purposes~